src/vegasflow/vflow.py

```diff
@@ -124,6 +124,32 @@ def _generate_random_array(rnds, divisions):
     return x, weights, ind_xn
 
 
+###### Auxiliary functions for the while loop
+@tf.function
+def while_check(bin_weight, *args):
+    """Checks whether the bin has enough weight
+    to beat the average"""
+    return bin_weight < ave_t
+
+@tf.function(
+    input_signature=[
+        tf.TensorSpec(shape=[], dtype=DTYPE),
+        tf.TensorSpec(shape=[], dtype=DTYPEINT),
+        tf.TensorSpec(shape=[], dtype=DTYPE),
+        tf.TensorSpec(shape=[], dtype=DTYPE),
+    ]
+)
+def while_body(bin_weight, n_bin, cur, prev):
+    """Fills the bin weight until it surpassed the avg
+    once it's done, returns the limits of the last bin"""
+    n_bin += 1
+    bin_weight += wei_t[n_bin]
```

Comment:
I don’t think this would work, wei_t is the resson this needs to be retraced a few times

Reply:
My bad. Maybe pass it as an arg, but things would be more complex then. Sorry.

Reply:
Yes, they would.

Did you test the code of the PR?